hack/tools/go.mod

```diff
@@ -7,12 +7,10 @@ require (
 	github.com/joelanford/go-apidiff v0.1.0
 	github.com/onsi/ginkgo v1.16.5
 	github.com/vmware/govmomi v0.23.1
-	k8s.io/code-generator v0.22.2
-	sigs.k8s.io/cluster-api/hack/tools v0.0.0-20211104153216-fb1f86267fed
-	sigs.k8s.io/controller-runtime/tools/setup-envtest v0.0.0-20211025141024-c73b143dc503
+	k8s.io/code-generator v0.27.2
+	sigs.k8s.io/cluster-api/hack/tools v0.0.0-20230720182410-9c0a017bd4e3
+	sigs.k8s.io/controller-runtime/tools/setup-envtest v0.0.0-20221201045826-d9912251cd81
```

Comment:
Had to bump this a bit to be compatible with the new hack/tools.

If we needed another reason to move everythign we can out of hack/tools/go.mod..